blockchain_connector/Dockerfile

```diff
@@ -103,66 +105,76 @@ COPY --from=openssl_image /usr/local/lib /usr/local/lib
 RUN ldconfig \
  && ln -s /etc/ssl/certs/* /usr/local/ssl/certs/
 
-ENV TCF_HOME=/project/avalon
-
 COPY VERSION /project/avalon/
 COPY ./bin /project/avalon/bin
-COPY ./common/cpp /project/avalon/common/cpp
-COPY ./common/python /project/avalon/common/python
-COPY ./common/crypto_utils /project/avalon/common/crypto_utils
-COPY ./sdk /project/avalon/sdk
-COPY ./blockchain_connector /project/avalon/blockchain_connector
 
-WORKDIR /project/avalon/common/cpp
+ENV TCF_HOME=/project/avalon
 
-RUN mkdir -p build \
-  && cd build \
-  && cmake .. -DUNTRUSTED_ONLY=1 \
-  && make
+COPY ./common/python /project/avalon/common/python
 
-#install avalon common module
 WORKDIR /project/avalon/common/python
 
 RUN echo "Building Avalon Common Python\n" \
-  && python setup.py bdist_wheel \
-  && echo "Installing Avalon Common Python" \
-  && pip3 install dist/*.whl
+ && make
 
-#install avalon crypto module
-WORKDIR /project/avalon/common/crypto_utils
 
-RUN echo "Building Avalon Common Crypto Python\n" \
-  && make \
-  && echo "Installing Avalon Common Crypto Python" \
-  && pip3 install dist/*.whl
 
+# -------------=== Avalon SDK build ===-------------
+
+#Build Avalon SDK intermediate docker image
+FROM python_image as avalon_sdk_image
+
+RUN apt-get update \
+ && apt-get install -y -q \
+    make
+
+COPY VERSION /project/avalon/
+COPY ./bin /project/avalon/bin
+COPY ./sdk /project/avalon/sdk
+
+ENV TCF_HOME=/project/avalon
 
 WORKDIR /project/avalon/sdk
+
 #install Avalon SDK  package.
 RUN echo "Building Avalon SDK\n" \
- && make \
- && echo "Installing Avalon SDK\n" \
- && make install
+ && make
 
 
-# Install fabric python sdk
-RUN pip3 install virtualenv \
-   && pip3 install git+https://github.com/hyperledger/fabric-sdk-py.git
-
-# install blockchain connector module
-RUN cd /project/avalon/blockchain_connector \
-   && python setup.py bdist_wheel \
-   && pip3 install dist/*.whl
 
 # Build final image and install modules
 FROM base_image as final_image
 
+RUN apt-get update \
+ && apt-get install -y -q \
+    git
+
+# Installing web3 for ethereum
+RUN pip3 install --upgrade json-rpc web3 py-solc-x nose2 eth-abi
+
+# Install specific version of solc to support versions <0.6.0
+RUN python3 -m solcx.install v0.5.15
+
+RUN pip3 install nest_asyncio
+
+COPY VERSION /project/avalon/
+COPY ./bin /project/avalon/bin
+COPY ./blockchain_connector /project/avalon/blockchain_connector
+
 ENV TCF_HOME=/project/avalon
 
 WORKDIR /project/avalon/blockchain_connector
 
-COPY --from=build_image /usr/local/ /usr/local
-COPY --from=build_image /root/.solcx /root/.solcx
+COPY --from=common_python_image /project/avalon/common/python/dist/*.whl dist/
+COPY --from=avalon_sdk_image /project/avalon/sdk/dist/*.whl dist/
+
+# Install fabric python sdk
+RUN pip3 install virtualenv \
+   && pip3 install git+https://github.com/hyperledger/fabric-sdk-py.git
+
+# install blockchain connector module
+RUN python setup.py bdist_wheel \
+   && pip3 install dist/*.whl
```

Comment:
Can we create a Makefile for blockchain connector module and execute `make` and `make install` command to install wheel

Reply:
ok